File: delivery-system-server/src/main/java/ru/delivery/system/common/JsonSerializer.java
package ru.delivery.system.common;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;
import java.text.SimpleDateFormat;

public class JsonSerializer {
    private static final ObjectMapper mapper = new ObjectMapper();

    static {
        mapper.setDateFormat(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss"));
    }

    public static <T> T toEntity(String json, Class<T> clazz) throws IOException {
        return mapper.readValue(json, clazz);
    }

    public static <T> T toEntity(String json, TypeReference<T> reference) throws IOException {
        return mapper.readValue(json, reference);
    }

    public static <T> String toJson(T entity) throws JsonProcessingException {
        mapper.setDateFormat(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss"));
        return mapper.writeValueAsString(entity);
    }
}

File: delivery-system-android/app/src/main/java/ru/delivery/system/android/utils/JsonSerializer.kt
package ru.delivery.system.android.utils

import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
import com.fasterxml.jackson.module.kotlin.readValue

class JsonSerializer {
    val mapper = jacksonObjectMapper()

    fun <T> toJson(entity: T) : String {
        return mapper.writeValueAsString(entity)
    }

    inline fun <reified T> toEntity(json : String) : T {
        return mapper.readValue(json)
    }

}